### DISCUSSION TITLE
Attachments

### DISCUSSION THREAD

**Discussion by jorge-leon** [django-helpdesk/django-helpdesk#1188] [Ideas]
Attachments use the default file storage with `MEDIA_ROOT`/`MEDIA_URL` settings.  So they depend completely on the setup of the respective settings.

By default, `MEDIA_URL` in Django is thought to be served statically by the web server in production for efficiency.

Most probably, ticket attachment will contain sensitive information and should only be served to authorized users.

Since Django 4.2, different storages can be declared in the global settings and then used in `model.FileField`/`model.ImageField`s.

One approach could be to:

1. Provide an authorized view for the django-helpdesk attachments and use it as default.
2. Provide a django-helpdesk storage and use it as default
3. Allow to specify the storage to be used for django-helpdesk attachments in the settings.

**Comment by uhurusurfa** [comment]
Actually irt is STATIC_ROOT that defines the statically served files so use of MEDIA_ROOT/MEDIA_URL is can be correct if you view the attachments to tickets as media files.
Currently there is no concept of media files in the app so using the MEDIA_XXX settings is acceptable in a standalone setup.

However it is reasonable to assume that separation of helpdesk files from other files in an app that uses helpdesk as a plugin.
I would lean towards option 3.
